sktime/forecasting/base/adapters/_statsmodels.py

```diff
@@ -100,6 +100,13 @@ def _predict(self, fh, X=None):
         # beginning of the training series when passing integers
         start, end = fh.to_absolute_int(self._y.index[0], self.cutoff)[[0, -1]]
 
+        # bug fix for evaluate function as test_plus_train indices are passed
+        # statsmodels exog must contain test indices only.
+        # For discussion see https://github.com/sktime/sktime/issues/3830
+        if X is not None:
+            ind_drop = self._X.index
+            X = X.loc[~X.index.isin(ind_drop)]
```

Comment:
not sure if that is correct bcz how will that work for in-sample predictions? Then you would drop the in-sample `X` values here also? Can you make sure that in-sample prediction still works with exog data?

Reply:
Thanks for the quick response @aiwalter.  I wasn't considering the requirement for in-sample predictions.  Good catch.  I'll review the code and my changes with this in mind and respond.

Regarding your comments:

> might be better to fix it in `_split` as you mentioned in the issue. 

I'll look into whether it is possible to fix in `_split`.  

> Not sure why the in `X.iloc[test_plus_train].sort_index()` the train data is appended to the test data. You would have to check if all tests are passing and ideally check why this was implemented like that. Might be that some forecasters need both as input to make predictions?

You are right, some transformers require `train_plus_test` to be passed for prediction.  @miraep8 responded to my issue and mentioned that this is related to [#2880](https://github.com/sktime/sktime/issues/2880).  Do you have any thoughts on this?